lib/easy_lock.h

```diff
@@ -43,6 +43,16 @@
 #define curl_simple_lock atomic_bool
 #define CURL_SIMPLE_LOCK_INIT false
 
+/* a clang-thing */
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
+#if (defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))) || \
```

Comment:
Clang also defines `__GNUC__`, so `defined(__GNUC__)` needs to instead be `(defined(__GNUC__) && !defined(__clang__))`.